rules/aip0231/message_rules.go

```diff
@@ -0,0 +1,143 @@
+// standard get request message field, but the latter one is not suggested.
+var namesField = &lint.MessageRule{
+	Name:   lint.NewRuleName("core", "0231", "request-message", "name-field"),
+	URI:    "https://aip.dev/231#request-message",
+	OnlyIf: isBatchGetRequestMessage,
+	LintMessage: func(m *desc.MessageDescriptor) (problems []lint.Problem) {
+		// Rule check: Establish that a name field is present.
+		names := m.FindFieldByName("names")
+		getReqMsg := m.FindFieldByName("requests")
+
+		// Rule check: Ensure that the names field is existed.
+		if names == nil && getReqMsg == nil {
+			problems = append(problems, lint.Problem{
+				Message:    fmt.Sprintf(`Message %q has no "names" field`, m.GetName()),
+				Descriptor: m,
+			})
+		}
+
+		// Rule check: Ensure that only the suggested names field is existed.
+		if names != nil && getReqMsg != nil {
+			problems = append(problems, lint.Problem{
+				Message:    fmt.Sprintf(`Message %q should delete "requests" field, only keep the "names" field`, m.GetName()),
+				Descriptor: getReqMsg,
+			})
+		}
+
+		// Rule check: Ensure that the names field is repeated.
+		if names != nil && !names.IsRepeated() {
+			problems = append(problems, lint.Problem{
+				Message:    `The "names"" field should be repeated`,
+				Descriptor: names,
+			})
+		}
+
+		// Rule check: Ensure that the names field is the correct type.
+		if names != nil && names.GetType() != builder.FieldTypeString().GetType() {
+			problems = append(problems, lint.Problem{
+				Message:    `"names" field on Batch Get Request should be a "string" type`,
+				Descriptor: names,
+			})
+		}
+
+		// Rule check: Ensure that the standard get request message field is repeated.
+		if getReqMsg != nil && !getReqMsg.IsRepeated() {
+			problems = append(problems, lint.Problem{
+				Message:    `The "requests" field should be repeated`,
+				Descriptor: getReqMsg,
+			})
+		}
+		// Rule check: Ensure that the standard get request message field is the correct type.
+		rightTypeName := fmt.Sprintf("Get%sRequest", inflect.ToSingular(m.GetName()[8:len(m.GetName())-7]))
+		if getReqMsg != nil && (getReqMsg.GetMessageType() == nil || getReqMsg.GetMessageType().GetName() != rightTypeName) {
+			problems = append(problems, lint.Problem{
+				Message:    fmt.Sprintf(`The "requests" field on Batch Get Request should be a %q type`, rightTypeName),
+				Descriptor: getReqMsg,
+			})
+		}
+		return
+	},
+}
+
+// The Batch Get response message should have resource field.
+var resourceField = &lint.MessageRule{
+	Name:   lint.NewRuleName("core", "0231", "response-message", "resource-field"),
+	URI:    "https://aip.dev/231#response-message",
+	OnlyIf: isBatchGetResponseMessage,
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		// the singular form the resource name, the first letter is Capitalized
+		resourceMsgName := inflect.ToSingular(m.GetName()[8 : len(m.GetName())-9])
```

Comment:
Consider to add a comment to explain `m.GetName()[8 : len(m.GetName())-9]`.

Reply:
Done.